### ISSUE TITLE
Question: SentenceTransformer vs AutoModel

### ISSUE THREAD

**Issue by dyukha** [UKPLab/sentence-transformers#1072]
First, thanks for the great library! For my purposes, I need to modify the architecture, and I first tried to reproduce `SentenceTransformer` behavior with `AutoModel` but encountered some issues. Namely, let `model_name` be a name of a pre-trained model (e.g. `sentence-transformers/paraphrase-mpnet-base-v2` or `sentence-transformers/bert-base-nli-cls-token`). Then is it true that for a batch of `texts`, the following:
```
return SentenceTransformer(model_name).encode(texts)
```
is equivalent to the following:
```
tokenizer = AutoTokenizer.from_pretrained(model_name)
tokenizer.model_max_length = 128
tokenized_texts = ...  # tokenize texts with the tokenizer
model = AutoModel.from_pretrained(model_name)
return model(**tokenized_texts).pooler_output  # and convert to numpy
```

I suspect that they are not the same, since:
* Without fine-tuning, `SentenceTransformer` produces more reasonable embeddings (e.g. when using `BERTopic`, for `AutoModel` significantly more sentences are unclustered)
* When fine-tuning, `SentenceTransformer` reaches good loss noticeably faster (as far as I can tell, `AutoModel` reaches the same loss but significantly later).

Do you know what I should change to reach similar behavior? I suspect that `SentenceTransformer` also uses some kind of better initialization, but I expected it to be the same in both cases. Thanks!

**Comment by nreimers** [comment]
model(**tokenized_texts).pooler_output returns either CLS token or some dense layer on top of the CLS token depending on the model architecture (like BERT, it returns the output of the next sentence prediction head).
In SBERT, usually mean pooling is used. 

Here you can find the code for mean pooling:
https://huggingface.co/sentence-transformers/bert-base-nli-mean-tokens

The initialization is not changed. It just uses the AutoModel under the hood.

**Comment by dyukha** [comment]
Thank you for the reply! Did I understand you correctly that SBERT essentially ignores the pooling layer of the model and always uses mean, and so `bert-base-nli-cls-token` essentially becomes `bert-base-nli-mean-token`? And the code is in `SentenceTransformer._load_auto_model(...)`.

**Comment by nreimers** [comment]
The pooling layer is always ignored.

The CLS models use the CLS token as embedding, the mean models use mean pooling

**Comment by dyukha** [comment]
Got it, thanks!
